File: Sudoku_Java_Shell/src/cspSolver/Assignment.java
package cspSolver;
/**
 * Assignments represent assigned variables and values associated with them.
 *   
 * @see LocalSolver
 */
public class Assignment implements Comparable<Assignment>{

	//===============================================================================
	// Properties
	//===============================================================================
	
	private Variable variable;
	private int value;
	private int score;
	
	//===============================================================================
	// Constructors
	//===============================================================================
	
	public Assignment(Variable v, int val)
	{
		this.variable = v;
		this.value = val;
	}
	
	public Assignment(Variable v, int val, int score)
	{
		this.variable = v;
		this.value = val;
		this.score = score;
	}
	
	//===============================================================================
	// Accessors
	//===============================================================================
	
	public Variable getVariable() {
		return variable;
	}

	public int getValue() {
		return value;
	}	
	
	public int getScore() {
		return score;
	}

	//===============================================================================
	// Modifiers
	//===============================================================================
	
	public void setValue(int val)
	{
		this.value = val;
	}
	
	public void setScore(int score)
	{
		this.score = score;
	}
	//===============================================================================
	// Comparable<Assignment>
	//===============================================================================
	
	@Override
	public int compareTo(Assignment a) {
		return this.score - a.score;
	}
	//===============================================================================
	// String Methods
	//===============================================================================
		
	public String toString()
	{
		StringBuilder sb = new StringBuilder();
		sb.append("Variable: " + variable.getName());
		sb.append("\tValue: " + value);
		sb.append("\tscore: " + score);
		
		return sb.toString();
		
	}
}

File: Sudoku_Cpp_Shell/src/Assignment.cpp
#include"Assignment.hpp"

Assignment::Assignment ( Variable* v, int val )
{
	variable = v;
	value = val;
}

Assignment::Assignment ( Variable* v, int val, int score )
{
	variable = v;
	value = val;
}

Variable* Assignment::getVariable ( void )
{
    return variable;
}

int Assignment::getValue ( void ) const
{

    return value;
}

int Assignment::getScore ( void ) const
{
    return score;
}

void Assignment::setValue ( int val )
{
    this->value = val;
}

void Assignment::setScore ( int score )
{
    this->score = score;
}

int Assignment::compareTo ( Assignment a )
{
    return this->score - a.score;
}

std::string Assignment::toString()
{
    std::stringstream ss;
    ss << "Variable: " << variable->getName();
    ss << "\tValue: " << this->value;
    ss << "\tscore: " << this->score;
    return ss.str();
}

